client/files.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// RemovePathOptions holds the options for a call to RemovePath.
+type RemovePathOptions struct {
+	// Path is the absolute path to be deleted (required).
+	Path string
+
+	// MakeParents, if true, specifies that any non-existent parent directories
+	// should be created. If false (the default), the call will fail if the
+	// directory to be created has at least one parent directory that does not
+	// exist.
+	Recursive bool
+}
+
+type removePathsPayload struct {
+	Action string            `json:"action"`
+	Paths  []removePathsItem `json:"paths"`
+}
+
+type removePathsItem struct {
+	Path      string `json:"path"`
+	Recursive bool   `json:"recursive"`
+}
+
+type fileResult struct {
+	Path  string       `json:"path"`
+	Error *errorResult `json:"error,omitempty"`
+}
+
+type errorResult struct {
+	Message string      `json:"message"`
+	Kind    string      `json:"kind,omitempty"`
+	Value   interface{} `json:"value,omitempty"`
+}
```

Comment:
There's going to be quite a lot of merge conflicts when we merge all of these. :-) But that's fine -- I think `fileResult` and `errorResult` should be general across all the file APIs.

Reply:
You're completely right (: I didn't find any alternative to this while working on multiple related functionalities at once, and didn't want any PR to be dependent on each other.

Reply:
Yeah, I think that's the right choice for review -- thanks!